pages/api/auth.ts

```diff
@@ -1,13 +1,42 @@
 import { NextApiRequest, NextApiResponse } from 'next';
+import { createAppExtension } from '@lib/appExtensions';
 import { encodePayload, getBCAuth, setSession } from '../../lib/auth';
 
 export default async function auth(req: NextApiRequest, res: NextApiResponse) {
     try {
         // Authenticate the app on install
         const session = await getBCAuth(req.query);
         const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+        const { access_token: accessToken, context } = session;
+
+        const storeHash = context?.split('/')[1] || '';
 
         await setSession(session);
+
+        /**
+         * For stores that do not have the app installed yet, create App Extensions when app is
+         * installed.
+         */
+
+        const requestBody = await createAppExtension();
+
+        const response = await fetch(
+            `https://${process.env.API_URL}/stores/${storeHash}/graphql`,
+            {
+                method: 'POST',
+                headers: {
+                    accept: 'application/json',
+                    'content-type': 'application/json',
+                    'x-auth-token': accessToken,
+                },
+                body: JSON.stringify(requestBody),
+            }
+        );
+        if (!response.ok) {
+            const error = await response.json();
+            throw new Error(error.message || 'Unknown error occurred');
```

Comment:
`'Unknown error occurred'` - We do know some things about the error. For example, I think a better message would describe to the app developer why their request failed. Something like that might be: <code>\`Response status code ${response.status} received from BigCommerce API when trying to create app extensions\`</code>

Reply:
Ahhh okay, that makes sense and a better description!